SaltSquareIO/sports-app-react-fe
# Registration API

[CLICKUP_TICKET_ID](https://app.clickup.com/t/86c0g7kwk)

## Summary of change

This PR finalizes the registration flow by sending the data to the `/auth/register` endpoint instead of printing out the values in the console. Additionally:

1. First name, last name and gender are sent in the request body with defaults if they are not provided by the user. If no first name is provided, "FName" is sent. For the last name, the default value is "LName". The gender is hardcoded to "DECLINE_TO_IDENTIFY". 
2. After a successful registration, the user is redirected to the Home page which is represented by a background image. 
3. If the email is already in use, an error message is displayed under the email input field. 
4. If there is a server error or another undefined failure, a modal with the message "Something went wrong, please try again!" is shown. Upon closing the modal, the user remains on the registration screen but all input fields are cleared.

## QA Notes

To test this PR, follow these instructions:

1. Test the registration flow by attempting to register with and without unrequired fields (first name, last name). 
2. Verify that the redirection to the home page happens after a successful registration.
3. Try registering with an already taken email to ensure the "User with this email already exists." error is shown under the email field. 
4. Simulate a server failure to test that a pop-up message "Something went wrong, please try again!" appears.  

## Screenshots

### Error message if the email is already taken 

<img width="505" alt="Screenshot 2024-10-24 at 10 10 44" src="https://github.com/user-attachments/assets/ffc26ce7-cbe1-4281-a643-1ca393801df9">

### Home Page 

<img width="1903" alt="Screenshot 2024-10-24 at 10 27 40" src="https://github.com/user-attachments/assets/49829a81-b22c-43ef-af0e-978fe95ab442">

### Pop-up message after a server failure or something else that is undefined

<img width="976" alt="Screenshot 2024-10-24 at 10 12 16" src="https://github.com/user-attachments/assets/9df8fb3b-f92e-4497-9a61-8064a51d2acb">
